src/response.rs

```diff
@@ -0,0 +1,438 @@
+//! Responses to [requests](crate::request).
+
+use core::{
+    ffi::{c_char, c_void, CStr},
+    ptr::NonNull,
+    time::Duration,
+};
+
+use crate::{
+    file,
+    framebuffer::{Framebuffer, RawFramebuffer},
+    memory_map,
+    paging::{Flags, Mode},
+    smp,
+};
+
+macro_rules! impl_base_fns {
+    () => {
+        /// Returns the revision of the response.
+        pub fn revision(&self) -> u64 {
+            self.revision
+        }
+    };
+}
+
+/// A response to a [bootloader info
+/// request](crate::request::BootloaderInfoRequest).
+#[repr(C)]
+pub struct BootloaderInfoResponse {
+    revision: u64,
+    name: *const c_char,
+    version: *const c_char,
+}
+unsafe impl Sync for BootloaderInfoResponse {}
+unsafe impl Send for BootloaderInfoResponse {}
+impl BootloaderInfoResponse {
+    impl_base_fns!();
+
+    /// Returns the name of the loading bootloader.
+    pub fn name(&self) -> &str {
```

Comment:
Should these 2 functions perhaps return a `Result` instead of possibly panicking?

Reply:
According to the spec, they're ASCII strings. Therefore, they must be valid UTF-8.